router/default.go

```diff
@@ -692,15 +682,55 @@ func (r *router) flushRouteEvents(evType EventType) ([]*Event, error) {
 		return nil, fmt.Errorf("failed listing routes: %s", err)
 	}
 
+	if r.options.Advertise == All {
+		// build a list of events to advertise
+		events := make([]*Event, len(routes))
+		for i, route := range routes {
+			event := &Event{
+				Type:      evType,
+				Timestamp: time.Now(),
+				Route:     route,
+			}
+			events[i] = event
+		}
+		return events, nil
+	}
+
+	// routeMap stores optimal routes per service
+	optimalRoutes := make(map[string]Route)
+
+	// go through all routes found in the routing table and collapse them to optimal routes
+	for _, route := range routes {
+		optimal, ok := optimalRoutes[route.Service]
+		if !ok {
+			optimalRoutes[route.Service] = route
```

Comment:
This should probably be service+network as we'll likely provide isolated networks